examples/msal-net/akvdotnet/TokenCredential.cs

```diff
@@ -22,9 +22,12 @@ public MyClientAssertionCredential()
         var tokenPath = Environment.GetEnvironmentVariable("AZURE_FEDERATED_TOKEN_FILE");
         var tenantID = Environment.GetEnvironmentVariable("AZURE_TENANT_ID");
 
-        _confidentialClientApp = ConfidentialClientApplicationBuilder.Create(clientID)
-                .WithClientAssertion(ReadJWTFromFS(tokenPath))
-                .WithTenantId(tenantID).Build();
+        _confidentialClientApp = ConfidentialClientApplicationBuilder
+                .Create(clientID)
+                .WithAuthority("https://login.microsoftonline.com/", tenantID) // for simplicity, this samples assume the app is in the public cloud
+                .WithClientAssertion(() => ReadJWTFromFS(tokenPath))          // make sure that ReadJWTFromFS always returns a fresh JWT 
```

Comment:
Can we do something similar to https://github.com/Azure/azure-workload-identity/blob/ffb15ec83c68d13b5343871f8c404012a45eadc7/examples/msal-go/token_credential.go#L66

Read from file system every 5 minutes instead of every time?

Reply:
Yes, done.